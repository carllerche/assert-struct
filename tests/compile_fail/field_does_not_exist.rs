use assert_struct::assert_struct;

#[derive(Debug)]
struct User {
    name: String,
    age: u32,
}

fn main() {
    let user = User {
        name: "Alice".to_string(),
        age: 30,
    };

    // This should NOT compile - field 'email' does not exist
    assert_struct!(user, User {
        name: "Alice",
        age: 30,
        email: "alice@example.com",
    });
}