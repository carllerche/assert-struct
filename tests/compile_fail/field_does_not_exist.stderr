error[E0026]: struct `User` does not have a field named `email`
  --> tests/compile_fail/field_does_not_exist.rs:19:9
   |
19 |         email: "alice@example.com",
   |         ^^^^^ struct `User` does not have this field
